### PR TITLE
feat: added no notifition yet message bar

### PR DESCRIPTION
**Pull request by sundasnoreen12**
[INF-1082](https://2u-internal.atlassian.net/browse/INF-1082)

**Description**

1. "No notification yet" message should be visible in the notification tray when there is no notification for the user.
    Figma: [Notifications](https://www.figma.com/file/1WsIDaf9Alfy2r8eNYj6vX/Notifications?type=design&node-id=678-15078&mode=design)
2. Added test case "Successfully showed No notification yet message when the notification tray is empty".

**Screenshots**
**Before**
Screenshot 2023-10-13 at 5 21 55 PM

**After**
image

### CODE REVIEW

**Review by abdullahwaheed — CHANGES_REQUESTED** [review]
Can you please add unit tests for the feature you are adding?

File: src/Notifications/notificationTabs.test.jsx [hunk 1]
```diff
@@ -72,20 +72,4 @@ describe('Notification Tabs test cases.', () => {
 
     expect(within(tabs[0]).queryByRole('status')).toBeInTheDocument();
   });
-
-  it('Successfully selected reminder tab.', async () => {
```
[the remark this answers is not in the document]
Comment (abdullahwaheed) [reply src/Notifications/notificationTabs.test.jsx:76]: don't remove this test, its decreasing [coverage](https://github.com/edx/frontend-component-header-edx/pull/493#issuecomment-1757499702)

**Review by abdullahwaheed — APPROVED** [review]

File: src/Notifications/index.jsx [hunk 2]
```diff
@@ -210,38 +211,40 @@ const Notifications = () => {
...
+                      className="icon-size-20 text-primary-500"
+                      data-testid="setting-icon"
+                      screenReaderText="preferences settings icon"
+                    />
+                  </Hyperlink>
+                </Popover.Title>
+              </div>
               <Popover.Content className="notification-content p-0">
-                <NotificationTabs />
+                <NotificationTabs popoverHeaderRef={headerRef} notificationRef={popoverRef} />
```
Comment (awais-ansari) [inline src/Notifications/index.jsx:247]: create a context to pass these ref to nested children.

**Review by awais-ansari — CHANGES_REQUESTED** [review]

File: src/Notifications/index.jsx [hunk 3]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable react/jsx-no-constructed-context-values */
```
Comment (awais-ansari) [inline src/Notifications/index.jsx:1]: try to remove disable eslint

**Review by awais-ansari — CHANGES_REQUESTED** [review]

**Review by awais-ansari — APPROVED** [review]

**Pull request merged**